src/modules/wallet/sagas.ts

```diff
@@ -157,6 +171,47 @@ function* handleConnectWalletSuccess() {
   }
 }
 
+function* handleSwitchNetworkRequest(action: SwitchNetworkRequestAction) {
+  const { chainId } = action.payload
+  const provider: Provider | null = yield call(getConnectedProvider)
+  try {
+    if (!provider) {
+      throw new Error('Could not get provider')
+    }
+    yield call([provider, 'request'], {
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: '0x' + chainId.toString(16) }]
+    })
+    yield put(switchNetworkSuccess(chainId))
+  } catch (switchError) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if (provider && switchError.code === 4902) {
+      try {
+        yield call([provider, 'request'], {
```

Comment:
This is something that I've discovered while reading the tests. As there's a test missing (the failing path of adding an Ethereum chain), the [Metamask docs](https://docs.metamask.io/guide/rpc-api.html#other-rpc-methods) say that the user might choose to switch to the other network after adding the network, but what if they don't? We'll be signaling a `switchNetworkSuccess` but we'll still be on the other network.

Reply:
good catch! added a check after adding the network to verify the network chainId indeed changed, and also added a test for that case.